---
title: "Lab 10"
author: "Ben Peters"
date: "November 16, 2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(vcfR)
```

```{r}
data(vcfR_example)
vcf
```
```{r}
strwrap(vcf@meta[1:7])
```
```{r}
queryMETA(vcf)
```
```{r}
queryMETA(vcf, element = 'DP')
```
```{r}
queryMETA(vcf, element = 'FORMAT=<ID=DP')
```
```{r}
head(getFIX(vcf))
```
```{r}
vcf@gt[1:6, 1:4]
```

## Exercises Part I
1. We can find out more about read.vcfR by running ?read.vcfR in the console.

2. By using the queryMETA command and telling it to display info on the AD element, like so:
```{r}
queryMETA(vcf, element = "AD")
```

3. The last few lines of any data can be viewed using 'tail', like so:
```{r}
tail(vcf@fix)
```
4. Knowing the quality of the variants could be useful for assessing and filtering out the less high-quality parts of the data, which could bias or otherwise complicate our analysis of the data, so in that sense it could be useful in some circumstances, but for much of the work we're doing here it appears somewhat unnecessary.

5. By using 'colnames' like so:
```{r}
colnames(vcf@gt)
```

```{r}
library('vcfR')
vcf <- read.vcfR("Grunwald/pinfsc50_filtered.vcf.gz")
```
```{r}
vcf
```
```{r}
x <- vcfR2genlight(vcf)
```
```{r}
x
```

```{r}
gt <- extract.gt(vcf, element = "GT")
gt[c(2,6,18), 1:3]
```
```{r}
t(as.matrix(x))[c(1,5,17), 1:3]
```
```{r}
library(adegenet)
```
```{r}
pop(x) <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
popNames(x)
```
```{r}
ploidy(x) <- 2
```
```{r}
x.dist <- dist(x)
```
```{r}
x.dist <- poppr::bitwise.dist(x)
```
```{r}
library(vcfR)

# Find the files.
vcf_file <- system.file("extdata", "pinf_sc50.vcf.gz", package = "pinfsc50")
dna_file <- system.file("extdata", "pinf_sc50.fasta", package = "pinfsc50")
gff_file <- system.file("extdata", "pinf_sc50.gff", package = "pinfsc50")

# Input the files.
vcf <- read.vcfR(vcf_file, verbose = FALSE)
dna <- ape::read.dna(dna_file, format = "fasta")
gff <- read.table(gff_file, sep="\t", quote="")

# Create a chromR object.
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=TRUE)
```
```{r}
chrom
```
```{r}
plot(chrom)
```
```{r}
chromoqc(chrom, dp.alpha = 66)
```
```{r}
chrom <- proc.chromR(chrom, verbose = TRUE)
```
```{r}
plot(chrom)
```
```{r}
chromoqc(chrom, dp.alpha = 66)
```
```{r}
vcf <- read.vcfR("Grunwald/pinfsc50_filtered.vcf.gz", verbose = FALSE)
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=FALSE)
chrom <- proc.chromR(chrom, verbose = FALSE)
chromoqc(chrom, dp.alpha = 66)
```
```{r}
head(chrom@var.info)
```
```{r}
head(chrom@win.info)
```
```{r}
library(vcfR)
data(vcfR_example)
pop <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
myDiff <- genetic_diff(vcf, pops = pop, method = 'nei')
knitr::kable(head(myDiff[,1:15]))
```
```{r}
knitr::kable(head(myDiff[,16:19]))
```
```{r}
knitr::kable(round(colMeans(myDiff[,c(3:9,16,19)], na.rm = TRUE), digits = 3))
```
```{r}
library(reshape2)
library(ggplot2)

dpf <- melt(myDiff[,c(3:8,19)], varnames=c('Index', 'Sample'), value.name = 'Depth', na.rm=TRUE)
```
```{r}
p <- ggplot(dpf, aes(x=variable, y=Depth)) + geom_violin(fill="#2ca25f", adjust = 1.2)
p <- p + xlab("")
p <- p + ylab("")
p <- p + theme_bw()
p
```
## Exercises Part II
1. 
```{r}
plot(getPOS(vcf), myDiff$Gprimest,  pch = 20, xlab = "", ylab = "", ylim = c(0, 1), xaxt = "n")
axis(side = 1, at = seq(0, 1e5, by = 1e4), labels = seq(0, 100, by = 10))
title(xlab='Genomic Position')
title(ylab = expression("G' ST"))
```
2. The population size is very small, posing an issue for calculating genetic distance between the populations due to the small sample size.

3. 
```{r}
chromoqc(chrom, dp.alpha = 66, xlim = c(100000, 150000))
```
4. 
```{r}
queryMETA(vcf)
```
```{r}
queryMETA(vcf, element = "InbreedingCoeff")
queryMETA(vcf, element = "MLEAC")
queryMETA(vcf, element = "MLEAF")
```

